metpy/calc/thermo.py

```diff
@@ -402,18 +402,27 @@ def lfc(pressure, temperature, dewpt, parcel_temperature_profile=None):
     x, y = find_intersections(pressure[1:], parcel_temperature_profile[1:],
                               temperature[1:], direction='increasing')
 
+    # Compute LCL for this parcel for future comparisons
+    this_lcl = lcl(pressure[0], temperature[0], dewpt[0])
+
     # The LFC could:
     # 1) Not exist
     # 2) Exist but be equal to the LCL
     # 3) Exist and be above the LCL
 
     # LFC does not exist or is LCL
     if len(x) == 0:
-        if np.all(_less_or_close(parcel_temperature_profile, temperature)):
+        # Find the part of the sounding above the LCL
+        p = pressure.m
+        # searchsorted function requires increasing order
+        k = len(p) - np.searchsorted(p[::-1], this_lcl[0])
+
+        # Is there any positive area above the LCL?
+        if np.all(_less_or_close(parcel_temperature_profile[k:], temperature[k:])):
```

Comment:
Instead of finding the actual index, could we do:
```python
        if np.all(_less_or_close(parcel_temperature_profile[pressure < this_lcl], temperature[pressure < this_lcl])):
```

Reply:
Ahh, the boolean indexing is much clearer. I have one thing to check, and then I'll push to the branch accordingly.